src/wp-includes/class-walker-nav-menu.php

```diff
@@ -170,9 +187,30 @@ public function start_el( &$output, $data_object, $depth = 0, $args = null, $cur
 		 * @param int      $depth        Depth of menu item. Used for padding.
 		 */
 		$id = apply_filters( 'nav_menu_item_id', 'menu-item-' . $menu_item->ID, $menu_item, $args, $depth );
-		$id = $id ? ' id="' . esc_attr( $id ) . '"' : '';
 
-		$output .= $indent . '<li' . $id . $class_names . '>';
+		$li_atts 		  = array();
+		$li_atts['class'] = ! empty( $class_names ) ? $class_names : '';
+		$li_atts['id']    = ! empty( $id ) ? $id : '';
+
+		/**
+		 * Filters the HTML attributes applied to a menu item's li element.
+		 *
+		 * @since 6.1.2
+		 *
+		 * @param array $li_atts {
+		 *     The HTML attributes applied to the menu item's `<li>` element, empty strings are ignored.
+		 *
+		 *     @type string $class        HTML CSS class attribute.
+		 *     @type string $id           ID html attribute.
+		 * }
+		 * @param WP_Post  $menu_item The current menu item object.
+		 * @param stdClass $args      An object of wp_nav_menu() arguments.
+		 * @param int      $depth     Depth of menu item. Used for padding.
+		 */
+		$li_atts 	   = apply_filters( 'nav_menu_item_attributes', $li_atts, $menu_item, $args, $depth );
+		$li_attributes = $this->build_atts($li_atts);
```

Comment:
```suggestion
		$li_attributes = $this->build_atts( $li_atts );
```
Should fix these PHPCS failures.